mypy.ini

```diff
@@ -0,0 +1,8 @@
+[mypy]
+plugins = pydantic.mypy
+
+[pydantic-mypy]
+init_forbid_extra = True
```

Comment:
We might need to remove this, but @QMalcolm can make that adjustment when it becomes an issue.

Reply:
Is this the `forbid_extra` which makes it so that extras during pydantic deserialization are complained about? Or does this `forbid_extra` do something else?

Reply:
When `init_forbit_extra = True`, mypy throws an error for the following case:

```
class ExampleObject(FrozenBaseModel):  # noqa: D
    field0: int

def some_func() -> None:  # noqa: D
    logger.error(ExampleObject(field0=0, field1=2))
```

Otherwise, mypy does not.